### PR TITLE
Upgrade django to 1.11.5

### PR DESCRIPTION
**Pull request by richtier**
[this task](https://uktrade.atlassian.net/browse/ED-2104)

note [`django-revproxy`](https://github.com/TracyWebTech/django-revproxy
) currently only officially supports up to `1.10`. 

Do not merge. Need to update `directory-validators` etc dependencies once they are ready.

### CODE REVIEW

File: enrolment/tests/test_widgets.py [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 from django import forms
+
 from enrolment import widgets
 
 
 def minify_html(html):
     return html.replace('  ', '').replace('\n', '')
 
 
+@pytest.mark.skip('<input attributes are ordered non-deterministically')
```
Comment (richtier) [inline enrolment/tests/test_widgets.py]: upgrading django changed behaviour of the HTML renderer: the HTML element attributes are no longer ordered. We can investigate using beautiful soup to better test this.

File: company/tests/test_views.py [hunk 2]
```diff
@@ -511,7 +526,9 @@ def test_case_study_create_api_success(
     }
     # django converts uploaded files to UploadedFile, which makes
     # `assert_called_once_with` tricky.
-    mock_create_case_study.assert_called_once_with(
```
Comment (konradko) [inline company/tests/test_views.py]: why change this? `assert_called_once_with` does exactly same and looks nicer

File: enrolment/tests/test_widgets.py (hunk 1 above)
Comment (konradko) [inline enrolment/tests/test_widgets.py]: so what's the plan here, this is gonna be fixed? Can't we just check that `attribute in input`?

File: enrolment/tests/test_widgets.py [hunk 3]
```diff
@@ -17,57 +21,47 @@ class MyTestForm(forms.Form):
...
-            <input id="id_field" name="field" type="checkbox" />
+            <input id="id_field" name="field" type="checkbox" required/>
             <label for="id_field">the label</label>
         </div>
     """
 
     assert minify_html(expected_html) in minify_html(str(form))
 
 
+@pytest.mark.skip('<input attributes are ordered non-deterministically')
```
Comment (konradko) [inline enrolment/tests/test_widgets.py]: ditto

File: company/tests/test_views.py (hunk 2 above)
Comment (richtier) [reply company/tests/test_views.py]: `assert_called_once_with` is too easy to get wrong - even if it looks correct.

in our codebase I've seen a few times this kind of things happening:

`assert mock_thing.called_once`

or

`mock_thing.assert_callled_once_with`

granted, I am the main cause of such things, but they are equally not caught in code review. The simplest solution is to treat `assert_called_once_with` with hostility

https://engineeringblog.yelp.com/2015/02/assert_called_once-threat-or-menace.html

File: enrolment/tests/test_widgets.py (hunk 1 above)
Comment (richtier) [reply enrolment/tests/test_widgets.py]: I will remove these tests, and re-add later when we have worked out a reliable way to test it.

the plan was to confirm the custom HTML rendering works.

**Comment by richtier** [comment]
I upgraded first to 1.10.x, then to 1.11.x in the same PR.

Upgrading is much easier than I expected because we have really good unit test coverage.

I've tested manually too.

Old style middleware is still supported in 1.11 but will be removed in 2. That means the app will continue to work until 2020 when 1.11 loses LTS forcing an upgrade to version 2.

I'll recheck the warnings. I saw nothing wrong, but will double check.

**Pull request merged**